### DISCUSSION TITLE
Disabling auto-formatting on C++

### DISCUSSION THREAD

**Discussion by hocman2** [zed-industries/zed#8219] [Feature Requests]
Is there a way to completely disable auto-formatting on save on C++ ?

I've tried these settings but they don't do anything
`  "languages": {
    "C++": {
      "soft_wrap": "none",
      "formatter": null
    }
  }`

**Comment by Moshyfawn** [comment]
```json
{
  "languages": {
    "C++": {
      "format_on_save": "off"
    }
  }
}
```

**Reply by hocman2** [reply]
thanks I feel stupid lol

**Reply by Moshyfawn** [reply]
No need to, the docs are a bit too vague for this stuff 👌

**Reply by SomeoneToIgnore** [reply]
At this point, seems that we might want to create another, formattting-related doc and have a bunch of common topics highlighted there.
Would be great to see a PR on the topic.

**Reply by Moshyfawn** [reply]
> At this point, seems that we might want to create another, formattting-related doc and have a bunch of common topics highlighted there.

The docs are complete enough, it's possible to find most of the info with the help of in-editor suggestions and sometimes past discussions on here / Discord. But it is still a research process, as the surface area of Zed is quite large these days.

> Would be great to see a PR on the topic.

I might be able to look into it, but there's still some dark areas for me.

**Reply by SomeoneToIgnore** [reply]
Thank you.

I use https://github.com/zed-industries/zed/discussions/7951#discussioncomment-8501391 as a reference and maybe the base idea there is right: we have to start with "how things are formatted" and boil down to configs in that new doc, while current state forces people to look through all formatting-related things and assemble the picture in their head.

**Reply by Moshyfawn** [reply]
That's great, thanks! I'll see if I can come up with something. Because you're right, an explanation baked by config examples is much better than having to parse the entire config docs every time you want to see if something you're trying to do is even possible.

**Reply by xpe** [reply]
Is there a typo above? Did something change? As of now, `"languages"` is not the correct key; `"language_overrides"` is.

**Comment by xpe** [comment]
As of May 22, 2024 `"languages"` is not the correct key; `"language_overrides"` is. So your `.zed/settings.json` should look like:

```json
{
  "language_overrides": {
    "C++": { "format_on_save": "off" }
  }
}
```

UPDATE on 2024-07-11: My above answer is out-of-date. See SomeoneToIgnore's comment below.

**Reply by maryisdead** [reply]
Is that still correct? `languages`worked for me, `language_overrides`did not.

**Reply by SomeoneToIgnore** [reply]
There's been a deprecation notice some time ago:

To quote https://zed.dev/releases/stable
```
Zed 0.141.2

06/26/2024
Breaking changes
SETTINGS DEPRECATION NOTICE

In your settings.json the following keys will need to be renamed:

    language_overrides -> languages
    copilot -> inline_completions
    show_copilot_suggestions -> show_inline_completions

Previously, we were supporting the old keys as an alias for the new keys, but this let to unintended/confusing failure modes. As a result we have deprecated the old keys.

See: [Settings Deprecation Notice 2024-06-17](https://github.com/zed-industries/zed/issues/13171) for more information.
```
